identity-core/src/common/timestamp.rs

```diff
@@ -165,6 +188,49 @@ impl Diff for Timestamp {
   }
 }
 
+/// A span of time.
+///
+/// This type is typically used to increment or decrement a [`Timestamp`].
+#[derive(Clone, Copy, Hash, PartialEq, Eq, PartialOrd, Ord, Deserialize, Serialize)]
+#[repr(transparent)]
+pub struct Duration(time::Duration);
+
+// copy a small subset of the functionality offered by time::Duration. By not copying everything it will be easier
+// to migrate to another implementation if that will be necessary in the future.
+// Do not allow negative Durations hence all constructors take u32 instead of i64, this makes it possible for us to
+// switch to relying on (for instance) std::time::Duration if we want to.
+impl Duration {
+  /// A second
+  pub const SECOND: Self = Self(time::Duration::SECOND);
+  /// A minute
+  pub const MINUTE: Self = Self(time::Duration::MINUTE);
+  /// An hour
+  pub const HOUR: Self = Self(time::Duration::HOUR);
+  /// A day
+  pub const DAY: Self = Self(time::Duration::DAY);
+  /// A week
+  pub const WEEK: Self = Self(time::Duration::WEEK);
+
+  /// Create a new [`Duration`] with the given amount of seconds.
+  pub const fn seconds(seconds: u32) -> Self {
+    Self(time::Duration::seconds(seconds as i64))
+  }
+  /// Create a new [`Duration`] with the given amount of minutes.
+  pub const fn minutes(minutes: u32) -> Self {
+    Self(time::Duration::minutes(minutes as i64))
+  }
+
+  /// Create a new [`Duration`] with the given amount of hours.
+  pub const fn hours(hours: u32) -> Self {
+    Self(time::Duration::hours(hours as i64))
+  }
+
+  /// Create a new [`Duration`] with the given amount of weeks.
+  pub const fn weeks(weeks: u32) -> Self {
+    Self(time::Duration::weeks(weeks as i64))
+  }
```

Comment:
Nit: use `number` instead of `amount`. Optional.

```suggestion
  /// Create a new [`Duration`] with the given number of seconds.
  pub const fn seconds(seconds: u32) -> Self {
    Self(time::Duration::seconds(seconds as i64))
  }
  /// Create a new [`Duration`] with the given number of minutes.
  pub const fn minutes(minutes: u32) -> Self {
    Self(time::Duration::minutes(minutes as i64))
  }

  /// Create a new [`Duration`] with the given number of hours.
  pub const fn hours(hours: u32) -> Self {
    Self(time::Duration::hours(hours as i64))
  }

  /// Create a new [`Duration`] with the given number of weeks.
  pub const fn weeks(weeks: u32) -> Self {
    Self(time::Duration::weeks(weeks as i64))
  }
```